collector_benchmark_test.go

```diff
@@ -0,0 +1,231 @@
+package ftdc
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/mongodb/mongo-go-driver/bson"
+)
+
+var result []byte
+
+func BenchmarkCollectorInterface(b *testing.B) {
+	for _, collect := range []struct {
```

Comment:
given that these are copy pasted it might make sense to add functions which return these fixtures for both the benchmarks and the tests. this would require defining some unexported types

Reply:
Could we just have a package level variable with this list so we only have to create it once? And the same for the list of tests created on line 68?

Reply:
package level variables are mutable, and I'd prefer to avoid confusing that in the future. the function is pretty clean.